feats/meta.py

```diff
@@ -33,22 +33,18 @@ class Definition:
     """
     """
 
-    def __init__(self, obj):
-        self.description = obj.__doc__
-        self.implementations: Dict[str, Implementation] = {}
-        self.annotations = defaultdict(list)
-        for key in dir(obj):
-            value = getattr(obj, key)
-            if key.startswith('_') or not callable(value):
-                continue
-            impl = Implementation(value)
-            self.implementations[impl.name] = impl
-            if hasattr(value, '_feats_annotations_'):
-                for annotation in value._feats_annotations_:
-                    self.annotations[annotation].append(impl)
-
-        if len(self.implementations) == 0:
+    def __init__(
+            self,
+            description: str,
+            implementations: Dict[str, Implementation],
+            annotations: Dict[str, list]):
+
+        if len(implementations) == 0:
             # TODO: Describe what an implementation needs
             raise ValueError(
                 "Definition did not contain at least one implementation"
             )
+
+        self.description = description
+        self.implementations = implementations
+        self.annotations = annotations
```

Comment:
We talked briefly about rolling annotations into implementations somehow, which I agree would probably clean this up a bit. I opted to not do it here to keep this PR a reasonable size for now, but if we want I can certainly look into that here as well. Otherwise I can open a ticket to look into it.